### ISSUE TITLE
Missing footnotes when aligning tables in dev version

### ISSUE THREAD

**Issue by meenaparam** [haozhu233/kableExtra#32]
I'm raising this issue in relation to this question on SO that you answered for me:

https://stackoverflow.com/questions/45378664/kableextra-rmarkdown-tables-aligning-grouping-row-labels-and-footnotes

I put an update of the problem in a comment but figured it would be easier for you to look at here. Basically now alignment of grouping rows works in the dev version, but the footnote text has disappeared totally rather than being aligned incorrectly. Not sure why this is happening as the superscript footnote numbers still appear in the table, it's just the footnote text itself that has disappeared. 

A further fix would be appreciated, if possible!

Here is the code to reproduce the error:

```{r}
library(devtools)
dev_mode(on = T)
install_github("haozhu233/kableExtra")

library(kableExtra)
library(knitr)

#' ### Example table

#' **Note that the actual text of footnotes is missing**

#+ cars-table, results='asis'
kable(mtcars[1:10, 1:2], format = "html", caption = "Group Rows", 
      col.names = c("MPG[note]", "CYL[note]")) %>%
    kable_styling("striped", full_width = F) %>%
    group_rows("Group 1", 4, 7) %>%
    group_rows("Group 2", 8, 10) %>% 
    add_footnote(c("Some footnote", "Some other footnote"))

dev_mode(on = F)
```

**Comment by haozhu233** [comment]
@meenaparam Thanks for reporting this bug! I recently made some changes to the HTML outputting method in most functions in kableExtra to solve problems like Stackoverflow: https://stackoverflow.com/questions/45262308/knit2wp-adds-text-with-kableextra. These fixes removed some linespaces, which caused bugs for `add_footnote`, which is the only HTML function that is using regular expression instead of `xml2` (because `add_footnote` is actually the first function I wrote for `kableExtra` and I was too lazy to make changes :P). 

It should be fairly easy to fix. For now, if you put `add_footnote` right after `kable`, the bug will be gone.
